tests/queries/0_stateless/03742_test_flattened_crash.sql

```diff
@@ -0,0 +1,25 @@
+DROP TABLE IF EXISTS test_flatten_nested_crash;
+CREATE TABLE test_flatten_nested_crash
+(
+    `id` UInt64,
+    `tenant` String,
+    `arr.id` Array(Nullable(UInt64)),
+    `arr.name` Array(Nullable(String)),
+    `arr.nested` Array(Tuple(a String, b Float64))
+)
+ENGINE = MergeTree
+ORDER BY (id)
+SETTINGS index_granularity = 8192;
+INSERT INTO test_flatten_nested_crash
+SELECT * FROM generateRandom(
+    '`id` UInt64,
+    `tenant` String,
+    `arr.id` Array(Nullable(UInt64)),
+    `arr.name` Array(Nullable(String)),
+    `arr.nested` Array(Tuple(a String, b Float64))', 1, 10
+) LIMIT 1000000;
+SELECT arr.nested FROM test_flatten_nested_crash ORDER BY arr.nested LIMIT 10000;
+ALTER TABLE test_flatten_nested_crash DROP COLUMN `arr.nested`;
+ALTER TABLE test_flatten_nested_crash ADD COLUMN `arr.nested` Array(Tuple(a String, b Float64));
+SELECT arr.nested FROM test_flatten_nested_crash ORDER BY arr.nested LIMIT 10000;
```

Comment:
Do we need the first SELECT to test the crash?
Is smth like `SELECT COUNT(*)` reproduces the issue too?
It feels a bit too much to commit 20k lines of random output 😄